### PR TITLE
Adding _shouldAnimate props

### PR DESCRIPTION
**Pull request by InventingWithMonster**

### CODE REVIEW

File: src/components/AnimateSharedLayout/types.ts [hunk 1]
```diff
@@ -34,25 +34,16 @@ export interface SharedLayoutProps {
...
-     * `SharedLayoutProps` will only run animate animations when this dependency changes.
+     * To improve performance we can pass use `_shouldAnimate` to disable this on particular render.
      *
-     * TODO: Potentially replace this with a shouldAnimate
+     * This is only intended for use within Framer.
      *
      * @internal
      */
-    _dependency?: any
+    _shouldAnimate?: any
```
Comment (hemlok) [inline src/components/AnimateSharedLayout/types.ts:46]: Should this be typed as a boolean now that it's being used as such?

File: src/components/AnimateSharedLayout/index.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ export class AnimateSharedLayout extends React.Component<SharedLayoutProps> {
      */
     syncContext: SharedLayoutSyncMethods = {
         ...createBatcher(),
-        syncUpdate: force => this.scheduleUpdate(force),
+        syncUpdate: () => this.scheduleUpdate(),
```
Comment (hemlok) [inline src/components/AnimateSharedLayout/index.tsx:51]: Was force doing something before or is this just cleanup of an unused prop?

File: src/motion/features/layout/Animate.tsx [hunk 3]
```diff
@@ -62,7 +62,12 @@ class Animate extends React.Component<AnimateProps> {
             ...config
         }: SharedLayoutAnimationConfig = {}
     ) => {
-        const { visualElement, layout } = this.props
+        const { visualElement, layout, _shouldAnimateLayout } = this.props
+
+        /**
+         * Early return if we've been instructed not to animate this render.
+         */
+        if (_shouldAnimateLayout === false) return this.safeToRemove()
```
Comment (hemlok) [inline src/motion/features/layout/Animate.tsx:70]: Is this the only place `_shouldAnimateLayout` is required in the refactored version of magic? Previously there were optimisations to use the lead/follow's `_shouldAnimate` for stack that enabled the optimization to work in Framer. Do we no longer need to support that?

File: src/components/AnimateSharedLayout/index.tsx (hunk 2 above)
Comment (InventingWithMonster) [reply src/components/AnimateSharedLayout/index.tsx:51]: Cleanup

File: src/motion/features/layout/Animate.tsx (hunk 3 above)
Comment (InventingWithMonster) [reply src/motion/features/layout/Animate.tsx:70]: Ah qui let me take a look

Comment (InventingWithMonster) [reply src/motion/features/layout/Animate.tsx]: Yeah you were right, just reimplemented

**Comment by hemlok** [comment]
@InventingWithMonster when you merge https://github.com/framer/motion/pull/678/files#submit-review, would you rebase this so we can test the example?

File: src/components/AnimateSharedLayout/animations.ts [hunk 4]
```diff
@@ -38,7 +38,9 @@ export function createCrossfadeAnimation(
     child: HTMLVisualElement,
     stack?: LayoutStack
 ): SharedLayoutAnimationConfig {
-    const config: SharedLayoutAnimationConfig = {}
+    const config: SharedLayoutAnimationConfig = {
+        shouldStackAnimate: stack ? stack.shouldStackAnimate() : true,
```
Comment (hemlok) [inline src/components/AnimateSharedLayout/animations.ts:42]: looks like we're not using this anywhere 👀

File: src/components/AnimateSharedLayout/types.ts [hunk 5]
```diff
@@ -34,25 +34,16 @@ export interface SharedLayoutProps {
...
 
     /**
-     * `transition` settings to use for the animate animation. Currently, only one transition
-     * can be defined for every animating value and every animating component.
+     * By default, `AnimateSharedLayout` will run layout animations on all children on every render,
+     * and on every child render.
      *
-     * @internal
-     */
-    _transition?: Transition
```
Comment (hemlok) [inline src/components/AnimateSharedLayout/types.ts:42]: Was this intended to be removed or a mistake?

**Pull request closed without merging**